File: Algorithms/flutter_algorithms/lib/testers/ackermann_function_tester.dart
import 'package:flutter_algorithms/testers/algorithm_tester.dart';
import 'package:flutter_algorithms/algorithms/ackermann_function.dart';

class AckermannFunctionTester extends AlgorithmTester {
  @override
  String testDescription(int testSize) {
      return "m = 3, n = ${3 + testSize}";
  }

  @override
  int test(int testSize) {
      int m = 3;
      int n = 3 + testSize;

      final stopwatch = Stopwatch()..start();
      AckermannFunction.call(m, n);
      stopwatch.stop();
      return stopwatch.elapsedMilliseconds;
  }
}

File: Algorithms/Android/app/src/main/java/com/example/androidalgorithms/testers/AckermannFunctionTester.java
package com.example.androidalgorithms.testers;

import com.example.androidalgorithms.algorithms.AckermannFunction;

public class AckermannFunctionTester extends AlgorithmTester {
    @Override
    public String testDescription(int testSize) {
        return String.format("m = %d, n = %d", 3, 1 + testSize);
    }

    @Override
    public double test(int testSize) {
        int m = 3;
        int n = 1 + testSize;

        long startTime = System.nanoTime();
        AckermannFunction.call(m, n);
        return System.nanoTime() - startTime;
    }
}